app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/MainActivityViewModel.kt

```diff
@@ -1,24 +1,20 @@
 package com.bottlerocketstudios.brarchitecture.ui
 
 import com.bottlerocketstudios.brarchitecture.data.buildconfig.BuildConfigProvider
-import com.bottlerocketstudios.brarchitecture.data.model.Repository
+import com.bottlerocketstudios.brarchitecture.data.converter.convertToUser
 import com.bottlerocketstudios.brarchitecture.data.repository.BitbucketRepository
+import com.bottlerocketstudios.brarchitecture.domain.models.Repository
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.map
 
 class MainActivityViewModel(val repo: BitbucketRepository, buildConfigProvider: BuildConfigProvider) : BaseViewModel() {
-    val selectedRepo: StateFlow<Repository> = MutableStateFlow(Repository())
-    val showToolbar: StateFlow<Boolean> = MutableStateFlow(false)
-    val title: StateFlow<String> = MutableStateFlow("")
+    val selectedRepo = MutableStateFlow(Repository(null, null, null, null, null, null, null))
+    val showToolbar = MutableStateFlow(false)
+    val title = MutableStateFlow("")
 
-    val avatarUrl = repo.user.map { it?.avatarUrl.orEmpty() }.groundState("")
+    val avatarUrl = repo.user.map { it?.convertToUser()?.avatarUrl.orEmpty() }.groundState("")
     val displayName = repo.user.map { it?.displayName.orEmpty() }.groundState("")
     val username = repo.user.map { it?.username.orEmpty() }.groundState("")
 
     val devOptionsEnabled = buildConfigProvider.isDebugOrInternalBuild
-
```

Comment:
I removed the use of the Set below because it was not type-safe.

Reply:
@VanyaHuamanBottleRocket I think this should be fine for now. Just note that if we need to do multiple things in the future we'll need to use a function at which point we should just use the `private val _foo: MutableStateFlow<Bar>` and `val foo: StateFlow<Bar>` approach is the `set` helper isn't address by then

cc: